lib/util/lint-result-cache.js

```diff
@@ -102,8 +108,21 @@ class LintResultCache {
     setCachedLintResults(filePath, result) {
         const fileDescriptor = this.fileEntryCache.getFileDescriptor(filePath);
 
-        if (!fileDescriptor.notFound) {
-            fileDescriptor.meta.results = result;
+        if (fileDescriptor && !fileDescriptor.notFound) {
+
+            // Serialize the result, except that we want to remove the file source if present.
+            const resultToSerialize = Object.assign({}, result);
+
+            /*
+             * Set result.source to null.
+             * In `getCachedLintResults`, if source is explicitly null, we will
+             * read the file from the filesystem to set the value again.
+             */
+            if (resultToSerialize.source) {
```

Comment:
Will this cause problems if the source is an empty string? That will cause this `if` check to fail even though the `source` property is present. It might be better to use an explicit `hasOwnProperty` check instead.